### ISSUE TITLE
Converting jp2 to png produces wrong image

### ISSUE THREAD

**Issue by zhutq** [ImageMagick/ImageMagick#2761]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am using the latest version of ImageMagick
- [x] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

I'm converting an jp2 image to png. The result image's color looks skewed.
`opj_decompress` v2.3.1 produces correct result.

### Steps to Reproduce

**Input image:**
https://raw.githubusercontent.com/zhutq/filehost/master/jp2/sample.jp2

**Converting with ImageMagick**
```
magick sample.jp2 sample.magick.png
```
The result looks wrong:
https://github.com/zhutq/filehost/blob/master/jp2/sample.magick.png

**Converting with OpenJPEG**
```
opj_decompress -i sample.jp2 -o sample.opj.png
```
The result looks good:
https://github.com/zhutq/filehost/blob/master/jp2/sample.opj.png

### System Configuration

- ImageMagick version:
```
Version: ImageMagick 7.0.10-34 Q16 x86_64 2020-10-09 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(4.5) 
Delegates (built-in): bzlib freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps tiff webp xml zlib
```

- Environment (Operating system, version and so on):
macOS Catalina
Version 10.15.6
ImageMagick was installed through brew

- Additional information:
`identify -list format` reports
```
JPT* JP2       rw-   JPEG-2000 File Format Syntax (2.3.1)
```
`opj_decompress ` also has been compiled against openjp2 library v2.3.1.

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT master branch @ https://github.com/ImageMagick/ImageMagick within a week or two.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ as soon as the patch is available.

**Comment by tanji** [comment]
Is there still a plan to fix this issue?

**Comment by dlemstra** [comment]
This has already been fixed and works correctly in the latest release of ImageMagick.

**Comment by tanji** [comment]
@dlemstra is this also fixed in 6.9? We use ImageMagick 6.9.10-23.

**Comment by dlemstra** [comment]
This was also fixed in ImageMagick 6 but you probably need to upgrade to a more recent version. The version you are using is from 2019 and this was fixed in 2020.
